File: test/languages/sample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Akihabara.Fixtures;

public enum ReviewState { Draft, Published }
public readonly record struct ReviewId(Guid Value);
public sealed record Review(ReviewId Id, string Title, double? Score, ReviewState State);

public interface IRepository<out T> { IEnumerable<T> All { get; } }

public sealed class ReviewService(IRepository<Review> repository)
{
    public const int DefaultLimit = 10;
    public event Action<Review>? Selected;

    public async Task<IReadOnlyList<string>> SummariesAsync(int limit = DefaultLimit)
    {
        await Task.Yield();
        return repository.All
            .Where(review => review.State is ReviewState.Published)
            .OrderByDescending(review => review.Score ?? 0)
            .Take(limit)
            .Select(review => $"{review.Title}: {review.Score:F1}")
            .ToArray();
    }

    public void Select(Review review) => Selected?.Invoke(review);
}

File: test/languages/sample.swift
import Foundation

enum ReviewState: String, Codable { case draft, published }

protocol IdentifiableRecord { associatedtype ID: Hashable; var id: ID { get } }

struct Review: IdentifiableRecord, Codable {
    let id: UUID
    var title: String
    var score: Double?
    var state: ReviewState

    var summary: String {
        let value = score.map { String(format: "%.1f", $0) } ?? "unrated"
        return "\(title): \(value)"
    }
}

actor ReviewStore {
    private var reviews: [UUID: Review] = [:]
    subscript(id: UUID) -> Review? { reviews[id] }
    func save(_ review: Review) { reviews[review.id] = review }
}

func published<T: Sequence>(_ values: T) -> [Review] where T.Element == Review {
    values.filter { $0.state == .published }.sorted { ($0.score ?? 0) > ($1.score ?? 0) }
}
